lib/ansible/galaxy/collection/galaxy_api_proxy.py

```diff
@@ -28,11 +28,21 @@
 class MultiGalaxyAPIProxy:
     """A proxy that abstracts talking to multiple Galaxy instances."""
 
-    def __init__(self, apis, concrete_artifacts_manager):
-        # type: (t.Iterable[GalaxyAPI], ConcreteArtifactsManager) -> None
+    def __init__(self, apis, concrete_artifacts_manager, offline=False):
+        # type: (t.Iterable[GalaxyAPI], ConcreteArtifactsManager, bool) -> None
         """Initialize the target APIs list."""
         self._apis = apis
         self._concrete_art_mgr = concrete_artifacts_manager
+        self._offline = offline  # Prevent all GalaxyAPI calls
+
+    @property
+    def is_offline_mode_requested(self):
+        return self._offline
+
+    def _assert_that_offline_mode_is_not_requested(self):  # type: () -> None
+        """Raises NotImplementedError if MultiGalaxyAPIProxy would make an GalaxyAPI call with --offline."""
```

Comment:
It's typically highly discouraged to refer to what caller would do under some uncontrolled circumstances.
```suggestion
        """Raises :class:`NotImplementedError` if offline mode has been requested."""
```

At some point, this method may start being called in other places, for example. By different callers, under different circumstances. And this explanation would be made outdated. PEP 257 states that
> The docstring for a function or method **should summarize its behavior** and document its arguments, return value(s), side effects, exceptions raised, and restrictions on when it can be called (all if applicable). Optional arguments should be indicated. It should be documented whether keyword arguments are part of the interface.

It's recommended to describe things where they happen. Otherwise, it'd be hard to trace them.

Reply:
```suggestion

```